ibis/common/validators.py

```diff
@@ -48,12 +59,41 @@ def __call__(self, arg, **kwargs):
 
         return self.validator(arg, **kwargs)
 
+    def __eq__(self, other):
+        return (
+            type(self) is type(other)
+            and self.validator == other.validator
+            and self.default == other.default
+        )
+
+
+class Variadic(Annotation):
+    """
+    Marker class for validating variadic arguments.
+    """
+
+    __slots__ = ('validator',)
+
+    def __init__(self, validator):
+        self.validator = validator
+
+    def __repr__(self):
+        return f"Variadic({self.validator!r})"
```

Comment:
Might as well use `self.__class__` here.

```suggestion
        return f"{self.__class__}({self.validator!r})"
```

Reply:
Not relevant of this PR's scope anymore.